### DISCUSSION TITLE
How to customise git-blame-echo so it works happily with inbuilt themese like modus-vivendi?

### DISCUSSION THREAD

**Discussion by ldeck** [magit/magit#4847] [Support]
As mentioned in the title, I'm currently using modus-vivendi with emacs 28.2. I've tried cycling to other themes.

When performing a git-blame-echo it seems the information is using the same foreground color as the background color. I can call magit-cycle-blame-style and see some things.

What might be a suggested for discovering any config that might be conflicting, or that should be customised to make it work happily?

**Comment by kyleam** [comment]
> it seems the information is using the same foreground color as the background color.

If that is indeed what's happening, then I think a fix would need to be on modus-vivendi's end.  But

 1) looking quickly at modus-themes.el, it seems to define the relevant faces

 2) the default `magit-blame-echo` style (determined by `magit-blame-echo-style`) is `lines`, which only shows a dividing line between chunks and then shows a message when you cross chunks (see the docstring of `magit-blame-styles`)

 3) testing on my end, I don't notice anything unexpected

If after reviewing the above, you still think something's off, could you post an image and point out what you think is unexpected?

**Comment by ldeck** [comment]
Here's how it looks for me...
<img width="655" alt="magit-blame-mystery" src="https://user-images.githubusercontent.com/65503/212793936-0b32c6c4-499c-4aef-8aa2-e3de2c073dcd.png">

**Reply by ldeck** [reply]
So @kyleam are you suggesting that the above are dividing lines? I assumed there was meant to be a title shown and info about the author and commit in the margin or something.

**Reply by kyleam** [reply]
Hmm, that's what you see with a fresh `magit-blame-echo`?

Fwiw here's what I see with modus-vivendi from a recent modus-themes commit (9721c05) on Emacs 28.2, running `magit-blame-echo` with the latest Magit commit and a vanilla config:

![emacs-28-latest-modus](https://user-images.githubusercontent.com/1297788/212798366-6a09b467-14b9-4760-a72f-3d930ab75b95.png)


That's what I mentioned above as not looking off/unexpected.

However, if I try with Emacs 28.2 and the bundled modus-vivendi, I have a hard time seeing the lines:

![emacs-28-builtin-modus](https://user-images.githubusercontent.com/1297788/212798393-7b0fc284-20ec-4d50-9716-973407d2c8c8.png)


So perhaps modus has had some improvements in this area.  It may be worth trying the latest release (4.0.1) to see if that improves things on your end.  (And if that doesn't help, it's probably worth trying to test things out with a minimal configuration and package set.)

**Reply by ldeck** [reply]
Yes @kyleam that's what I see running `M-x magit-blame-echo`.
Just not sure if there's something in my [config](https://github.com/ldeck/nix-home/blob/1343b8a580fa506454e8ef07169de0f277b692fc/lib/defaults/emacs.nix) that's interfering with the presentation of blame in magit.
I'm currently used the bundled theme; so you're suggesting that the [ones from melpa](https://melpa.org/#/modus-themes) might give a better result?
